packages/image_picker/image_picker/ios/Classes/FLTPHPickerSaveImageToPathOperation.m

```diff
@@ -82,51 +84,63 @@ - (void)start {
           if ([image isKindOfClass:[UIImage class]]) {
-            __block UIImage *localImage = image;
-            PHAsset *originalAsset =
-                [FLTImagePickerPhotoAssetUtil getAssetFromPHPickerResult:self.result];
-
-            if (self.maxWidth != (id)[NSNull null] || self.maxHeight != (id)[NSNull null]) {
-              localImage = [FLTImagePickerImageUtil scaledImage:localImage
-                                                       maxWidth:self.maxWidth
-                                                      maxHeight:self.maxHeight
-                                            isMetadataAvailable:originalAsset != nil];
-            }
-            __block NSString *savedPath;
-            if (!originalAsset) {
-              // Image picked without an original asset (e.g. User pick image without permission)
-              savedPath =
-                  [FLTImagePickerPhotoAssetUtil saveImageWithPickerInfo:nil
-                                                                  image:localImage
-                                                           imageQuality:self.desiredImageQuality];
-              [self completeOperationWithPath:savedPath];
-            } else {
-              [[PHImageManager defaultManager]
-                  requestImageDataForAsset:originalAsset
-                                   options:nil
-                             resultHandler:^(
-                                 NSData *_Nullable imageData, NSString *_Nullable dataUTI,
-                                 UIImageOrientation orientation, NSDictionary *_Nullable info) {
-                               // maxWidth and maxHeight are used only for GIF images.
-                               savedPath = [FLTImagePickerPhotoAssetUtil
-                                   saveImageWithOriginalImageData:imageData
-                                                            image:localImage
-                                                         maxWidth:self.maxWidth
-                                                        maxHeight:self.maxHeight
-                                                     imageQuality:self.desiredImageQuality];
-                               [self completeOperationWithPath:savedPath];
-                             }];
-            }
+            [self processImage:image];
           }
         }];
   } else {
     [self setFinished:YES];
   }
 }
 
+- (void)processImage:(UIImage *)localImage API_AVAILABLE(ios(14)) {
+  PHAsset *originalAsset = [FLTImagePickerPhotoAssetUtil getAssetFromPHPickerResult:self.result];
+
+  if (self.maxWidth != (id)[NSNull null] || self.maxHeight != (id)[NSNull null]) {
+    localImage = [FLTImagePickerImageUtil scaledImage:localImage
+                                             maxWidth:self.maxWidth
+                                            maxHeight:self.maxHeight
+                                  isMetadataAvailable:originalAsset != nil];
+  }
+  if (!originalAsset) {
```

Comment:
Since you're already moving this, please flip the condition and the blocks. This:
```
if (foo) {
  // foo case
} else {
  // !foo case
}
```

is generally much easier to read than:
```
if (!foo) {
  // !foo case
} else {
  // !!foo case, which is foo case
}
```